File: Tutor Demos/Week 8/Java/src/Seller.java
import javafx.application.Application;
import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.control.TextField;
import javafx.scene.layout.HBox;
import javafx.scene.text.Text;
import javafx.stage.Stage;

import java.text.DecimalFormat;

public class Seller extends Application {
    public static void main(String[] args) {
        launch(args);
    }

    private Label albumNameLbl;
    private TextField amountTf;
    private Button buyBtn;
    private Label profitLbl;
    private Text profitTxt;

    double SALE_PRICE = 34.99;

    @Override
    public void start(Stage stage) {

        albumNameLbl = new Label("Serenity");
        amountTf = new TextField("1");
        buyBtn = new Button("Buy");
        profitLbl = new Label("Profit: ");
        profitTxt = new Text("0");

        buyBtn.setOnAction(event -> sell());

        HBox hbox = new HBox(albumNameLbl, amountTf, buyBtn, profitLbl, profitTxt);
        hbox.setAlignment(Pos.CENTER);
        hbox.setSpacing(10);
        hbox.setPadding(new Insets(5));

        stage.setScene(new Scene(hbox));
        stage.setTitle("Seller");
        stage.show();
    }

    private int getAmount() {
        return Integer.parseInt(amountTf.getText());
    }

    private double getCurrentProfit() {
        return Double.parseDouble(profitTxt.getText());
    }

    private void sell() {
        double sale = getAmount() * SALE_PRICE;
        profitTxt.setText(formatted(sale+getCurrentProfit()));
    }

    private String formatted(double amount) {
        return new DecimalFormat(("#0.##")).format(amount);
    }
}

File: Tutor Demos/Week 8/Python/seller.py
from tkinter import *

class Seller:
    SALE_PRICE = 34.99
    def __init__(self, root):
        self.root = root
        self.album_lbl = Label(self.root, text="Serenity")
        self.amount_entry = Entry(self.root)
        self.buy_btn = Button(self.root, text="Buy", command=self.sell)
        self.profit_lbl = Label(self.root, text="Profit: ")
        self.profit_txt = Label(self.root, text="0")

        self.album_lbl.pack(side=LEFT, padx=5)
        self.amount_entry.pack(side=LEFT, padx=5)
        self.buy_btn.pack(side=LEFT, padx=5)
        self.profit_lbl.pack(side=LEFT, padx=5)
        self.profit_txt.pack(side=LEFT, padx=5)

    def get_amount(self):
        return float(self.amount_entry.get())

    def get_current_profit(self):
        return float(self.profit_txt.cget("text"))

    def sell(self):
        sale = self.get_amount() * Seller.SALE_PRICE
        total_profit = self.get_current_profit() + sale
        self.profit_txt.config(text=f"{total_profit:.2f}")

if __name__ == "__main__":
    root = Tk()
    root.title("Seller")
    Seller(root)
    root.mainloop()